recorder/src/main/cpp/profile_writer.cc

```diff
@@ -259,7 +264,13 @@ ProfileSerializingWriter::ProfileSerializingWriter(jvmtiEnv* _jvmti, ProfileWrit
 }
 
 ProfileSerializingWriter::~ProfileSerializingWriter() {
-    if (cpu_samples_flush_ctr != 0) flush();
+    std::vector<PerfCtx::TracePt> user_ctxs;
+    reg.user_ctxs(user_ctxs);
+    auto last_ctx_id_reported = next_ctx_id;
```

Comment:
Yes, good point, I thought about it, but couldn't come up with a better name, so gave-up at that time.

Will fix this.